### ISSUE TITLE
False warning no-shadow

### ISSUE THREAD

**Issue by csicky** [vuejs/eslint-plugin-vue#502]
**Tell us about your environment**

* **ESLint Version:**
4.19.1
* **eslint-plugin-vue Version:**
"eslint-plugin-vue": "^4.0.0",
* **Node Version:**
8.9.4
**Please show your full configuration:**

```

// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true
  },
  // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
  // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
  extends: ['plugin:vue/recommended', 'airbnb-base'],
  // required to lint *.vue files
  plugins: ['vue'],
  // check if imports actually resolve
  settings: {
    'import/resolver': {
      webpack: {
        config: 'build/webpack.base.conf.js'
      }
    }
  },
  // add your custom rules here
  rules: {
    // don't require .vue extension when importing
    'import/extensions': [
      'error',
      'always',
      {
        js: 'never',
        vue: 'never'
      }
    ],
    // disallow reassignment of function parameters
    // disallow parameter object manipulation except for specific exclusions
    'no-param-reassign': [
      'error',
      {
        props: true,
        ignorePropertyModificationsFor: [
          'state', // for vuex state
          'acc', // for reduce accumulators
          'e' // for e.returnvalue
        ]
      }
    ],
    // allow optionalDependencies
    'import/no-extraneous-dependencies': [
      'error',
      {
        optionalDependencies: ['test/unit/index.js']
      }
    ],
    'space-before-function-paren': [
      1,
      {
        anonymous: 'never',
        named: 'never',
        asyncArrow: 'never'
      }
    ],
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'comma-dangle': 0,
    semi: 0,
    'import/extensions': [1, { js: 'never' }, { vue: 'never' }],
    'no-param-reassign': 0,
    'no-unused-vars': 1,
    'one-var': 0,
    'no-bitwise': 0,
    'prefer-const': 0,
    'no-mixed-operators': 0,
    'no-plusplus': 0,
    'no-alert': 0,
    'no-var': 1,
    'max-len': 0,
    'no-console': 0,
    'no-unused-expressions': [1, { allowShortCircuit: true, allowTernary: true }],
    'arrow-parens': 0,
    'spaced-comment': 0,
    'no-nested-ternary': 0,
    'object-shorthand': 1,
    'arrow-body-style': 1,
    'dot-notation': 1,
    camelcase: 1,
    'no-multiple-empty-lines': 1,
    'import/first': 1,
    'prefer-template': 0,
    'prefer-arrow-callback': 1,
    'vue/max-attributes-per-line': 0,
    'vue/html-indent': 0,
    'vue/attribute-hyphenation': 0,
    'vue/attributes-order': 0,
    'vue/html-self-closing': 0,
    'vue/no-multi-spaces': 1,
    'vue/html-quotes': 1,
    'vue/v-on-style': 1,
    'vue/mustache-interpolation-spacing': 1,
    'vue/require-prop-types': 1,
    'vue/v-bind-style': 1,
    'vue/require-default-prop': 0,
    'vue/order-in-components': [
      1,
      {
        order: [
          'el',
          'name',
          'parent',
          'functional',
          ['delimiters', 'comments'],
          ['components', 'directives', 'filters'],
          'extends',
          'mixins',
          'inheritAttrs',
          'model',
          ['props', 'propsData'],
          'data',
          'computed',
          'methods',
          'LIFECYCLE_HOOKS',
          'watch',
          ['template', 'render'],
          'renderError'
        ]
      }
    ]
  }
}



```

**What did you do? Please include the actual source code causing the issue.**

```vue

             /* this warns object-shorthand  - correct */
            this.contactsDeleted.map(uuid => ({
              type: 'delete_contact',
              uuid: uuid
            }))


             /* this warns no-shadow - incorrect */
            this.contactsDeleted.map(uuid => ({
              type: 'delete_contact',
              uuid
            }))


```

**What did you expect to happen?**
I expect to not receive the `no-shadow` warning because I am not shadowing anything there.
Eslint plugin believes the shorthand notation to be shadowing the `uuid` param.

**What actually happened? Please include the actual, raw output from ESLint.**
http://eslint.org/docs/rules/no-shadow  'uuid' is already declared in the upper scope
  src\components\distributors\DistributorForm.vue:171:38
              this.contactsDeleted.map(uuid => ({

**Comment by csicky** [comment]
Actually now I see this is the wrong place to report this :) `no-shadow` is not part of `eslint-plugin-vue`.
